### PR TITLE
Add crystal survey 2017 results post

### PR DESCRIPTION
**Pull request by sdogruyol**
Please give your feedback :pray:

### CODE REVIEW

File: _posts/2017-11-03-2017-crystal-survey-2017-results.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+  "0.23.* %56.9, Not sure %22.4, Master %6.4")
+(hover for more info)
+
+Crystal developers tend to follow the latest stable release.
+
+![No %60.7, Yes, but it happens rarely %29.8, Yes, and it happens oftern %9.4](https://cdn.pbrd.co/images/GPdzxCW.png
+  "No %60.7, Yes, but it happens rarely %29.8, Yes, and it happens oftern %9.4")
+(hover for more info)
+
+We try to keep breaking changes as minimum as possible. More than %60 of Crystal developers don't have any changes to their code when upgrading to a new version.
```
Comment (RX14) [inline _posts/2017-11-03-2017-crystal-survey-2017-results.md]: This is pretty much a lie, we perform breaking changes all the time and with not much more reason than the change makes it better.

Comment (sdogruyol) [reply _posts/2017-11-03-2017-crystal-survey-2017-results.md]: Not everyone uses `HTTP::Handler`s and while maintaining Kemal for such a long time I haven't had many breaking changes after `0.20.0`. Just some minor stuff and compared to other pre-1.0  and heck even some 1.0+ languages it really feels minimum.

Comment (RX14) [reply _posts/2017-11-03-2017-crystal-survey-2017-results.md]: But that we *try* to keep breaking changes to a minimum is I personally think a lie. We break things for minor API niceties like removing `post_form`. And i think that's a good thing but I don't want to lie about it.

An interesting statistic would be the pie chart about breaking changes limited to people who say they have 10000+ line codebases.

File: _posts/2017-11-03-2017-crystal-survey-2017-results.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+- Microservices
+- CPU performance intensive apps
+- CLI tools
+- Background workers
+- Big Data processing
+- Ruby replacement
+
+Some very interesting ones
+
+- Genome analysis
```
Comment (CaDs) [inline _posts/2017-11-03-2017-crystal-survey-2017-results.md]: that was me ^^

**Comment by mverzilli** [comment]
Hi Serdar! Thanks for taking the time to run the survey and put together a first draft ^_^. I think this is a great starting point. However, we need to iterate a lot on each section to justify the value of the blogpost: right now it's little more than a screenshot of each part and a very superficial description of each set of results. If we can't provide some insight for each section, we should probably just link to the results and let people arrive to their own conclusions. I'll try to suggest some enhancements.

**Comment by sdogruyol** [comment]
Thanks @mverzilli, I'd be happy to hear how to enhance this resultset :+1:

File: _posts/2017-11-03-2017-crystal-survey-2017-results.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+We've received ***757*** responses, thanks a lot to everyone who took the time to respond.  
+
+We've tried our best to read through all of your comments and suggestions. Without further ado, let's take a look at the results.
+
+
+![%19.9 Google or other search engines, %19.4 HackerNews, %15.6 Reddit %10.9 Twitter](https://cdn.pbrd.co/images/GP30DmW.png
+  "%19.9 Google or other search engines, %19.4 HackerNews, %15.6 Reddit %10.9 Twitter")
+(hover for more info)
+
+We wanted to learn how did you discover Crystal, it's interesting to see that a lot our respondents find Crystal from search engines. HackerNews, Reddit and Twitter shows the power of social media.
```
Comment (mverzilli) [inline _posts/2017-11-03-2017-crystal-survey-2017-results.md]: We wanted to learn how you discovered Crystal, and were surprised to find that a lot of you arrived at Crystal through search engines. On the other hand, we confirmed how important HackerNews and Reddit to divulge young and small projects such as Crystal. Every now and then, someone from the community posts something interesting about Crystal, the post makes it to HN and Reddit front pages and in the next few days we notice a wave of new users and contributors on the different community channels.

File: _posts/2017-11-03-2017-crystal-survey-2017-results.md [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+
+# Using Crystal
+
+![%47,2 Yes, I' using Crystal for hobby projects, %32.1 No, but I' planning to use Crystal, %10.7 No, I'm not using Crystal, %9.9 Yes, I'm Using Crystal in production](https://cdn.pbrd.co/images/GP33qmx.png
+"%47,2 Yes, I' using Crystal for hobby projects, %32.1 No, but I' planning to use Crystal, %10.7 No, I'm not using Crystal, %9.9 Yes, I'm Using Crystal in production")
+(hover for more info)
+
+It's amazing to see a lot people already using Crystal. Nearly half of our respondents use Crystal in for hobby projects. Another interesting point is that nearly 1 out of 10 are running in production.
```
Comment (mverzilli) [inline _posts/2017-11-03-2017-crystal-survey-2017-results.md]: This is not a "verbatim" suggestion, I'm just tossing some lines of thought here. Let's continue asking ourselves what surprised us and what was kind of expected in each section. Some facts to highlight here: 

- There's a considerable number of people (42.9%) who's not using Crystal at all but still took the time to complete the survey, why?
- We sort of confirmed our intuition that at this point Crystal is mostly a hobby language.
- We're *psyched* that almost 10% of the respondents are already using Crystal in production. We want them to raise their hands, tell us how it's going for them and invite them to post in our blog.
- There's a deeper story that we see in these numbers, which is consistent with what we hear in informal discussions about Crystal: the community is eager to start using the language in production, but is waiting for more concrete signals from the project. A considerable part of the community doesn't even mind about that and trusts the ecosystem enough to take the leap.

File: _posts/2017-11-03-2017-crystal-survey-2017-results.md [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+"%63.1 Less than 1.000, %23 1000 to 10000 lines, %11.5 Not sure, %2.+ 10000 to 100000 line")
+(hover for more info)
+
+Most of the current Crystal projects are less than 1000 lines, However there are also 17 projects which are in between 10000 and 100000 lines.
+
+![%46.6 Yes, %32.9 Not sure, %11.9 No, %8.9 Not applicable](https://cdn.pbrd.co/images/GP39tRi.png
+  "%46.6 Yes, %32.9 Not sure, %11.9 No, %8.9 Not applicable")
+(hover for more info)
+
+Crystal is a beauty, we know that a lot of you are excited to use it for your daily work :)
```
Comment (mverzilli) [inline _posts/2017-11-03-2017-crystal-survey-2017-results.md]: We can combine these results with the previous couple of questions and hypothesize that for many, Crystal is good enough to knock out small (and performant) scripts that help at work without entailing the risk of a fully fledged, mission critical production system.

File: _posts/2017-11-03-2017-crystal-survey-2017-results.md [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+- Big Data processing
+- Ruby replacement
+
+Some very interesting ones
+
+- Genome analysis
+- HPC
+- Embedded GUI applications
+
+## How could we help make Crystal more accepted at your company?
```
Comment (mverzilli) [inline _posts/2017-11-03-2017-crystal-survey-2017-results.md]: Some possible conclusions to add here:

- No surprises in the list: we know all of these are the main gaps to fill and we're happy to be aligned with the widest community.
- Not many "vanity features" in the list. The language seems to provide a rich enough feature set for the community. The gaps to bridge are more related to the ecosystem (platform support, tooling, libraries and human resources) than to the language itself (the notable exception here being "parallelism").

File: _posts/2017-11-03-2017-crystal-survey-2017-results.md (hunk 1 above)
Comment (mverzilli) [reply _posts/2017-11-03-2017-crystal-survey-2017-results.md]: I'd say that we're mindful of breaking changes but at this point we value cleaner APIs over backwards compatibility. And I think that's true, see for example the pre-0.24 discussions around `Time`.

Comment (ysbaddaden) [reply _posts/2017-11-03-2017-crystal-survey-2017-results.md]: In the end no breaking change has been a pain to me: shards and my libraries seldom break, like once every 6 or 12 months, and are quickly fixed with a tiny patch. Even the 0.16 overhaul on types (all instance variables must be typed) was painless.

Comment (RX14) [reply _posts/2017-11-03-2017-crystal-survey-2017-results.md]: Fir enough, but that's not true for anyone and neither was the statement about the *quality* of breaking changes, just the *quantity*.

**Comment by RX14** [comment]
@sdogruyol this has been a while, will you get a chance to work on this soon?

**Comment by sdogruyol** [comment]
hey everyone, I've updated the post 👍

**Comment by sdogruyol** [comment]
Bump, let's publish this before it's christmas :pray:

**Comment by mverzilli** [comment]
I'll merge it as is and re-read it to see how it feels. If there's any minor thing to correct I'll just do it, if there's anything non-trivial I'll send a new PR. Thanks for pushing Serdar!

**Comment by sdogruyol** [comment]
Thank you @mverzilli :+1:

**Pull request merged**